subsys/net/lib/dns/resolve.c

```diff
@@ -432,6 +432,10 @@ static void release_query(struct dns_pending_query *pending_query)
 	if (busy == 0) {
 		/* All done. */
 		pending_query->cb = NULL;
+		/* set query also to NULL to prevent rogue call to
+		 * invoke_query_callback() causing a crash
+		 */
+		pending_query->query = NULL;
```

Comment:
Couldn't `invoke_query_callback()` simply verify the `pending_query->cb` before calling it instead? IMO it'd be cleaner solution.

Reply:
True, will add also but I still think it's weird not setting the query to null when the request is done.

Reply:
You could say the same about any other `struct dns_pending_query` member, that it should be cleared upon query completion. This is not the case though, as all the fields are re-initialized when a new query is issued. The only reason the query pointer is cleared in the condition below is that the author assigned it a special, "control" role if the query could not be released immediately, as the comment explains.

Reply:
Ok, added the check in `invoke_query_callback()`